pkg/network/infraconfigurators/masquerade_test.go

```diff
@@ -119,4 +131,218 @@ var _ = Describe("Masquerade infrastructure configurator", func() {
+				mockVML3Config(*masqueradeConfigurator, ifaceName, inPodBridge, additionalIPProtocol...)
+				mockNATNetfilterRules(handler, *dhcpConfig, backend, additionalIPProtocol...)
+
+				Expect(masqueradeConfigurator.PreparePodNetworkInterface()).To(Succeed())
+			},
+				table.Entry("NFTables backend on an IPv4 cluster", nft),
+				table.Entry("IPTables backend on an IPv4 cluster", ipTables),
+				table.Entry("NFTables backend on a dual stack cluster", nft, iptables.ProtocolIPv6),
+				table.Entry("IPTables backend on a dual stack cluster", ipTables, iptables.ProtocolIPv6),
+			)
+		})
+	})
 })
+
+func mockVML3Config(configurator MasqueradePodNetworkConfigurator, podIface string, inPodBridge *netlink.Bridge, optionalIPProtocol ...iptables.Protocol) {
+	protocols := getProtocols(optionalIPProtocol...)
+	hasIPv6Config := len(protocols) > 1
+	mockedHandler := configurator.handler.(*netdriver.MockNetworkHandler)
+	mockedHandler.EXPECT().IsIpv6Enabled(podIface).Return(hasIPv6Config, nil).Times(2) // once on create bridge, another on prepare pod network
+
+	for _, l3Protocol := range protocols {
+		gatewayAddr := configurator.vmGatewayAddr
+		if l3Protocol == iptables.ProtocolIPv6 {
+			gatewayAddr = configurator.vmGatewayIpv6Addr
+		}
+		mockedHandler.EXPECT().AddrAdd(inPodBridge, gatewayAddr).Return(nil)
+	}
+}
+
+func mockNATNetfilterRules(handler *netdriver.MockNetworkHandler, dhcpConfig cache.DHCPConfig, netfilterBackend netfilterBackend, optionalIPProtocol ...iptables.Protocol) {
+	getNFTIPString := func(proto iptables.Protocol) string {
+		ipString := "ip"
+		if proto == iptables.ProtocolIPv6 {
+			ipString = "ip6"
+		}
+		return ipString
+	}
+
+	for _, proto := range getProtocols(optionalIPProtocol...) {
+		vmIP := dhcpConfig.IP.IP.String()
+		gwIP := dhcpConfig.AdvertisingIPAddr.String()
+		if proto == iptables.ProtocolIPv6 {
+			vmIP = dhcpConfig.IPv6.IP.String()
+			gwIP = dhcpConfig.AdvertisingIPv6Addr.String()
+		}
+		mockNetfilterBackend(handler, proto, netfilterBackend, getNFTIPString(proto), vmIP, gwIP)
+	}
+}
+
+func mockNetfilterBackend(handler *netdriver.MockNetworkHandler, proto iptables.Protocol, backendType netfilterBackend, nftIPString string, vmIP string, gwIP string) {
+	handler.EXPECT().ConfigureIpForwarding(proto).Return(nil)
+
+	switch backendType {
+	case nft:
+		handler.EXPECT().NftablesLoad(proto).Return(nil)
+		handler.EXPECT().HasNatIptables(proto).Return(true).Times(0)
+		handler.EXPECT().HasNatIptables(proto).Return(false).Times(0)
+		mockNFTablesBackend(handler, proto, nftIPString, vmIP, gwIP)
+	case ipTables:
+		handler.EXPECT().NftablesLoad(proto).Return(fmt.Errorf("nft not found"))
+		handler.EXPECT().HasNatIptables(proto).Return(true)
+		mockIPTablesBackend(handler, proto, nftIPString, vmIP, gwIP)
+	}
+}
+
+func mockNFTablesBackend(handler *netdriver.MockNetworkHandler, proto iptables.Protocol, nftIPString string, vmIP string, gwIP string) {
```

Comment:
Move all this helpers as closure under Describe so we are sure they are not reused by _test.go files.

Reply:
I'm making these a struct in a couple of follow up commits so I can avoid passing a gazillion repeated arguments between calls.

Reply:
> Move all this helpers as closure under Describe so we are sure they are not reused by _test.go files.

and I also don't understand what's the motivation for anyone inside of this pkg to re-use something that only mocks NFT / iptables.